### ISSUE TITLE
[2.x] Update fails for debian packaged version 7.3.22-1

### ISSUE THREAD

**Issue by sceadara** [phpcq/phpcq#49]
```
In VersionParser.php line 176:
                                     
  [UnexpectedValueException]         
  Invalid version string "7.3.22-1"  
                                     

Exception trace:
  at /home/sceadara/applications/phpcq/vendor/composer/semver/src/VersionParser.php:176
 Composer\Semver\VersionParser->normalize() at /home/sceadara/applications/phpcq/src/Platform/PlatformRequirementChecker.php:32
 Phpcq\Platform\PlatformRequirementChecker::Phpcq\Platform\{closure}() at /home/sceadara/applications/phpcq/src/Platform/PlatformRequirementChecker.php:50
 Phpcq\Platform\PlatformRequirementChecker->isFulfilled() at /home/sceadara/applications/phpcq/src/Repository/Repository.php:118
 Phpcq\Repository\Repository->matchesPlatformRequirements() at /home/sceadara/applications/phpcq/src/Repository/Repository.php:100
 Phpcq\Repository\Repository->findMatchingVersions() at /home/sceadara/applications/phpcq/src/Repository/Repository.php:62
 Phpcq\Repository\Repository->getTool() at /home/sceadara/applications/phpcq/src/Repository/RemoteRepository.php:47
 Phpcq\Repository\RemoteRepository->getTool() at /home/sceadara/applications/phpcq/src/Repository/RepositoryHasToolTrait.php:14
 Phpcq\Repository\RemoteRepository->hasTool() at /home/sceadara/applications/phpcq/src/Repository/RepositoryPool.php:35
 Phpcq\Repository\RepositoryPool->getTool() at /home/sceadara/applications/phpcq/src/ToolUpdate/UpdateCalculator.php:69
 Phpcq\ToolUpdate\UpdateCalculator->calculateDesiredTools() at /home/sceadara/applications/phpcq/src/ToolUpdate/UpdateCalculator.php:57
 Phpcq\ToolUpdate\UpdateCalculator->calculate() at /home/sceadara/applications/phpcq/src/Command/UpdateCommand.php:43
 Phpcq\Command\UpdateCommand->calculateTasks() at /home/sceadara/applications/phpcq/src/Command/AbstractUpdateCommand.php:107
 Phpcq\Command\AbstractUpdateCommand->doExecute() at /home/sceadara/applications/phpcq/src/Command/AbstractCommand.php:102
 Phpcq\Command\AbstractCommand->execute() at /home/sceadara/applications/phpcq/vendor/symfony/console/Command/Command.php:258
 Symfony\Component\Console\Command\Command->run() at /home/sceadara/applications/phpcq/vendor/symfony/console/Application.php:911
 Symfony\Component\Console\Application->doRunCommand() at /home/sceadara/applications/phpcq/vendor/symfony/console/Application.php:264
 Symfony\Component\Console\Application->doRun() at /home/sceadara/applications/phpcq/vendor/symfony/console/Application.php:140
 Symfony\Component\Console\Application->run() at /home/sceadara/applications/phpcq/phpcq:50
```

```
$ php -v
PHP 7.3.22-1+0~20200909.67+debian10~1.gbpdd7b72 (cli) (built: Sep  9 2020 06:55:18) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.22, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.3.22-1+0~20200909.67+debian10~1.gbpdd7b72, Copyright (c) 1999-2018, by Zend Technologies
```

**Comment by discordier** [comment]
Worth to mention, not only `phpversion()` appears to yield an totally wrong version string (`7.3.22-1`) but also the PHP extensions yield the full string for some extensions `7.3.22-1+0~20200909.67+debian10~1.gbpdd7b72` which causes the version parser to throw the exception.

**Comment by dmolineus** [comment]
- Behaviour is documented in the user mentioned on php.net: https://www.php.net/manual/en/function.phpversion.php#119607
- Composer fixes it https://github.com/composer/composer/blob/5df1797d20c6ab1eb606dc0f0d76a16ba57ddb7f/src/Composer/Repository/PlatformRepository.php#L96-L102

**Comment by dmolineus** [comment]
Fixed with 20703c2.
